app/admin/actions.ts

```diff
@@ -0,0 +1,194 @@
+"use server"
+
+import { eq } from "drizzle-orm"
+import { authUsers } from "drizzle-orm/supabase"
+import { nanoid } from "nanoid"
+import { revalidatePath } from "next/cache"
+import { redirect } from "next/navigation"
+import { z } from "zod"
+
+import { db } from "@/db"
+import { apiAuthTokens, teams } from "@/db/schema"
+import { hashToken } from "@/lib/auth/hash-token"
+import { createClient } from "@/utils/supabase/server"
+
+const API_KEY_MANAGER_ROLE = "api_key_manager"
+
+const loginSchema = z.object({
+  email: z.string().email(),
+  password: z.string(),
+})
+
+export async function login(_prevState: unknown, formData: FormData) {
+  const validatedFields = loginSchema.safeParse({
+    email: formData.get("email"),
+    password: formData.get("password"),
+  })
+
+  if (!validatedFields.success) {
+    return {
+      errors: validatedFields.error.flatten().fieldErrors,
+    }
+  }
+
+  const { email, password } = validatedFields.data
+
+  const supabase = await createClient()
+
+  const { error } = await supabase.auth.signInWithPassword({
+    email,
+    password,
+  })
+
+  if (error) {
+    return {
+      errors: {
+        email: ["Invalid credentials"],
+      },
+    }
+  }
+
+  revalidatePath("/admin", "layout")
+  redirect("/admin")
+}
+
+const userSchema = z.object({
+  email: z.string().email(),
+  name: z.string(),
+})
+
+export async function createUser(_prevState: unknown, formData: FormData) {
+  const validatedFields = userSchema.safeParse({
+    email: formData.get("email"),
+    name: formData.get("name"),
+  })
+
+  if (!validatedFields.success) {
+    return {
+      errors: validatedFields.error.flatten().fieldErrors,
+    }
+  }
+
+  const { email, name } = validatedFields.data
+
+  try {
+    const supabase = createClient()
+
+    // check if logged user is using the correct role
+    const {
+      data: { user },
+    } = await supabase.auth.getUser()
+
+    if (user?.role !== API_KEY_MANAGER_ROLE) {
+      return {
+        errors: {
+          email: ["Unauthorized"],
+        },
+      }
+    }
+
+    // create user
+    const { data, error } = await supabase.auth.admin.createUser({
+      email,
+      email_confirm: true,
+    })
+
+    if (error) {
+      console.error("error creating user", error)
+      return {
+        errors: {
+          email: ["Error creating user"],
+        },
+      }
+    }
+
+    // update team name
+    await db
+      .update(teams)
+      .set({
+        name: name,
+      })
+      .where(eq(teams.id, data.user.id))
```

Comment:
teams.id and user.id are always the same?

Reply:
Correct. They are the same uuid <https://github.com/ethproofs/ethproofs/blob/dev/db/schema.ts#L183>. When a user is created, a new team row is added as well.